src/blockdata/block.rs

```diff
@@ -335,7 +337,7 @@ impl Block {
                 // Expand the push to exactly 8 bytes (LE).
                 let mut full = [0; 8];
                 full[0..b.len()].copy_from_slice(b);
-                Ok(util::endian::slice_to_u64_le(&full))
+                Ok(u64::from_le_bytes(full))
             }
```

Comment:
Looks like this could be:

```rust
let bytes = b.try_into()
    .map_err(|_| Bip34Error::UnexpectedPush(b.to_vec())))
    .map(|bytes| u64::from_le_bytes(*bytes));
```

Remove the extra match arm below and change `_` into `Instruction::Op(_)`.